ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/support/ArticleParser.java

```diff
@@ -62,16 +65,20 @@
             if (isArticleOrCategory(nextKey)) {
                 JSONObject returnedObject = evaluatingObject.optJSONObject(nextKey);
                 if (returnedObject != null) {
+                    urlParams.add(nextKey);
                     switch (returnedObject.optString(PRIMARY_TYPE_TAG)) {
                         case ORDER_FOLDER_TAG:
-                            articles.addAll(jsonObjectHandler(returnedObject));
+                            jsonObjectHandler(returnedObject, urlParams);
+                            // This param has been used and needs to be removed before looping.
 
                             break;
                         case PAGE_TAG:
-                            articles.add(parseArticleObject(returnedObject));
+                            articles.add(parseArticleObject(returnedObject, urlParams));
+                            // This param has been used and needs to be removed before looping.
 
                             break;
                     }
+                    urlParams = new ArrayList<>();
```

Comment:
resetting the `urlParams` here will lose any parent paths that were traversed already.

Reply:
using a `Uri` object you won't need to reset it after processing anymore.

Reply:
I had to reset the urlParams because I can have more than one Category or Article in the same Iteration of Keys.  So I am reseting the value before I loops to the next set of Keys.

Reply:
unfortunately the sequence of my comments wasn't preserved. I think using a `List` to keep track of the recursive parsing path is highly error prone, and how you have it right now will still produce errors.

Reply:
Yeah I didn't like using the list either but wasn't sure how to handle it another way.  I will try working on using a URI object.  To see if I can get it working with that way.